cdap-app-fabric/src/main/java/co/cask/cdap/app/services/DefaultStreamBatchWriter.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.app.services;
+
+import co.cask.cdap.api.data.stream.StreamBatchWriter;
+import co.cask.cdap.api.data.stream.StreamWriteException;
+import co.cask.cdap.common.io.ByteBuffers;
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.nio.ByteBuffer;
+
+/**
+ * Implementation of {@link StreamBatchWriter}.
+ */
+public class DefaultStreamBatchWriter implements StreamBatchWriter {
+
+  private final HttpURLConnection connection;
+  private final String stream;
```

Comment:
I think you should store Id.Stream here. Better for debugging, since it more uniquely identifies the stream (it includes namespace).